drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/docks/KogitoEditorDock.java

```diff
@@ -0,0 +1,114 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.kogito.client.docks;
+
+import java.util.Collection;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.google.gwt.core.client.GWT;
+import org.jboss.errai.ioc.client.api.ManagedInstance;
+import org.kie.workbench.common.widgets.client.docks.AuthoringEditorDock;
+import org.kie.workbench.common.widgets.client.docks.WorkbenchDocksHandler;
+import org.uberfire.client.workbench.docks.UberfireDock;
+import org.uberfire.client.workbench.docks.UberfireDockPosition;
+import org.uberfire.client.workbench.docks.UberfireDocks;
+import org.uberfire.mvp.Command;
+import org.uberfire.mvp.PlaceRequest;
+
+@ApplicationScoped
+public class KogitoEditorDock implements AuthoringEditorDock {
+
+    protected UberfireDocks uberfireDocks;
+    protected ManagedInstance<WorkbenchDocksHandler> installedHandlers;
+    protected String authoringPerspectiveIdentifier = null;
+    protected WorkbenchDocksHandler activeHandler = null;
+    protected UberfireDock[] activeDocks;
+
+    @Inject
+    public KogitoEditorDock(final UberfireDocks uberfireDocks,
+                            final ManagedInstance<WorkbenchDocksHandler> installedHandlers) {
+        this.uberfireDocks = uberfireDocks;
+        this.installedHandlers = installedHandlers;
+    }
+
+    @PostConstruct
+    public void initialize() {
+        // Initializing the handlers
+        installedHandlers.iterator().forEachRemaining(handler -> {
+            Command initCommand = () -> setActiveHandler(handler);
+            handler.init(initCommand);
+        });
+    }
+
+    @Override
+    public boolean isSetup() {
+        return authoringPerspectiveIdentifier != null;
+    }
+
+    @Override
+    public void setup(String authoringPerspectiveIdentifier, PlaceRequest defaultPlaceRequest) {
+        this.authoringPerspectiveIdentifier = authoringPerspectiveIdentifier;
+    }
+
+    @Override
+    public void show() {
+        GWT.log(this + " show");
+    }
+
+    @Override
+    public void hide() {
+        GWT.log(this + " hide");
+    }
+
+    @Override
+    public void expandAuthoringDock(UberfireDock dockToOpen) {
+        uberfireDocks.show(UberfireDockPosition.EAST, authoringPerspectiveIdentifier);
+        if (dockToOpen != null) {
+            uberfireDocks.open(dockToOpen);
+        }
+    }
+
+    protected void setActiveHandler(WorkbenchDocksHandler handler) {
+        // If there's an active handler let's check if it should refresh docks
+        if (activeHandler != null && activeHandler.equals(handler) && !activeHandler.shouldRefreshDocks()) {
```

Comment:
Can be simplified by `Objects.equals(activeHandler, handler)`

Reply:
@jomarko Done!